### ISSUE TITLE
[css-anchor-position] The "acceptable anchor" algo seems broken

### ISSUE THREAD

**Issue by jwatt** [w3c/csswg-drafts#11030]
The [acceptable anchor](https://drafts.csswg.org/css-anchor-position-1/#acceptable-anchor-element) text contains the following:

>- possible anchor is painted strictly before positioned el, aka one of the following is true:
>   - (snip)
>   - Both elements are [in the same top layer](https://drafts.csswg.org/css-position-4/#in-the-same-top-layer) but have different [containing blocks](https://drafts.csswg.org/css-display-4/#containing-block), and positioned el’s containing block is an ancestor of possible anchor’s containing block in the [containing block chain](https://drafts.csswg.org/css-display-4/#containing-block-chain), aka one of the following:
>     - positioned el’s [containing block](https://drafts.csswg.org/css-display-4/#containing-block) is the viewport, and possible anchor’s containing block isn’t.
>     - positioned el’s [containing block](https://drafts.csswg.org/css-display-4/#containing-block) is the [initial containing block](https://drafts.csswg.org/css-display-4/#initial-containing-block), and possible anchor’s containing block is generated by an element,
>     - both elements' [containing blocks](https://drafts.csswg.org/css-display-4/#containing-block) are generated by elements, and positioned el’s containing block is an ancestor in the [flat tree](https://drafts.csswg.org/css-scoping-1/#flat-tree) to that of possible anchor’s containing block.

I don't think some of these cases are correct.

For example:

> - positioned el’s [containing block](https://drafts.csswg.org/css-display-4/#containing-block) is the [initial containing block](https://drafts.csswg.org/css-display-4/#initial-containing-block), and possible anchor’s containing block is generated by an element,

seems to say that this should work:

```
<body>
  <div class="anchored" style="position: absolute;"></div>
  <div style="position: absolute;">
    <div class="anchor"></div>  
  </div>
</body>
```

And:

>    - both elements' [containing blocks](https://drafts.csswg.org/css-display-4/#containing-block) are generated by elements, and positioned el’s containing block is an ancestor in the [flat tree](https://drafts.csswg.org/css-scoping-1/#flat-tree) to that of possible anchor’s containing block.

seems to say that this should work:

```
<body>
  <div style="position: absolute;">
    <div class="anchored" style="position: absolute;"></div>
    <div style="position: absolute;">
      <div class="anchor"></div>  
    </div>
  </div>
</body>
```

In other words, in these two cases we sometimes still require that "`possible anchor` is earlier in [flat tree](https://drafts.csswg.org/css-scoping-1/#flat-tree) order than `positioned el`."

This seems...hard to spec.

**Comment by jwatt** [comment]
See also #11029

**Comment by jwatt** [comment]
It's really unfortunate that the term `absolutely positioned` is [formally understood](https://drafts.csswg.org/css2/#absolutely-positioned) to mean "absolutely positioned" or "fixed positioned", because I think we want to be able to say `absolutely positioned` in places here and **not** mean fixed positioned.

**Comment by tabatkins** [comment]
No, we don't want to distinguish abspos vs fixpos in that way, because a fixpos can *act* like an abspos given certain ancestor styles. What you actually care about is whether the containing block comes from the viewport, or something else, and that's the distinction I'm making in the algo.

That said, let me look at your actual feedback. ^_^

1. Yeah, you're right, the anchor there is *transitively* relying on the ICB, and after the abspos. That'll need fixing.

2. Yeah same deal.

But it's not about flat tree - in both of those cases, the flat tree is identical to the DOM tree, because there's no shadows involved. I'll just need to do some finessing about ancestors in the abspos containing block tree.

**Comment by tabatkins** [comment]
All right, (2) is solved by restoring the text I'd accidentally deleted earlier, and restored when fixing #11170. I solved (1) by just copying the same text over, as the exact same condition applies.
